modules/infra/datapath/loop_input.c

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr.h>
+#include <gr_control_input.h>
+#include <gr_graph.h>
+#include <gr_ip4_datapath.h>
+#include <gr_ip6_datapath.h>
+#include <gr_loopback.h>
+#include <gr_trace.h>
+
+#include <rte_graph_worker.h>
+
+static control_input_t control_to_loopback_input;
+
+control_input_t loopback_get_control_id(void) {
+	return control_to_loopback_input;
+}
+
+enum {
+	IP_LOCAL,
+	IP_OUTPUT,
+	IP6_LOCAL,
+	IP6_OUTPUT,
+	IP_NO_ROUTE,
+	IP6_NO_ROUTE,
+	EDGE_COUNT,
+};
+
+static uint16_t loopback_input_process(
+	struct rte_graph *graph,
+	struct rte_node *node,
+	void **objs,
+	uint16_t nb_objs
+) {
+	struct rte_mbuf *mbuf;
+	rte_edge_t edge;
+
+	for (unsigned i = 0; i < nb_objs; i++) {
```

Comment:
if ng_objs is uint16_t, i should be uint16_t but not unsigned. It just helps to support the consistency.

Reply:
good catch. Will send a new revision today.